cmd/destroy/destroy.go

```diff
@@ -162,6 +163,41 @@ func Destroy(ctx context.Context) *cobra.Command {
 				options.Namespace = okteto.Context().Namespace
 			}
 
+			// when option --all the cmd will destroy everything at the namespace and return
+			if options.DestroyAll {
+				if !okteto.Context().IsOkteto {
+					return errors.New("option `--all` is not available for non-Okteto clusters. Learn more: https://www.okteto.com/docs/self-hosted/")
+				}
+
+				okClient, err := okteto.NewOktetoClient()
+				if err != nil {
+					return err
+				}
+
+				if err := okClient.Namespaces().DestroyAll(ctx, options.Namespace); err != nil {
+					return err
+				}
+
+				stop := make(chan os.Signal, 1)
+				signal.Notify(stop, os.Interrupt)
+				exit := make(chan error, 1)
+				go func() {
+					err := okClient.Stream().DestroyAllLogs(ctx, options.Namespace)
+					if err != nil {
+						oktetoLog.Warning("destroy all logs cannot be streamed due to connectivity issues")
+						oktetoLog.Infof("destroy all logs cannot be streamed due to connectivity issues: %v", err)
+					}
+					exit <- err
+				}()
+				select {
+				case <-stop:
+					oktetoLog.Infof("CTRL+C received, exit")
+					return nil
+				case err := <-exit:
+					return err
+				}
```

Comment:
I don't think this logic has to be here. As I see, the structure of this (and other commands) are:
* Options/flags validation
* Running some pre configuration (mainly load the context)
* And then, command logic (usually through an struct like `destroyCommand`

In this case, we are including part of the command logic in the place where we are validating flags.

I would keep the logic of validating if the context is an Okteto one, but the rest, I would move it to:
* Within the function `runDestroy`
* Another function in the `destroyCommand` struct and in the line `217` include a conditional like

```
if options.DestroyAll {
  err = c.runDestroyAll(ctx, options)
} else { 
  err = c.runDestroy(ctx, options)
}
```

At the end, this logic belongs to `destroyCommand`.

Also, we wouldn't be tracking the destroy as we are doing in the line 218. Should we track it in this case too in an special way?

Reply:
I've done this change, also regarding tracking, i think this should be tracked separatly, or at least detect a different event as is a different use?

An other question that i just came up, i added includeVolumes as default true, wondering if we also want to add the flag -v and based on that?